bpf/tests/nat_test.h

```diff
@@ -99,3 +101,58 @@ void test_snat_v4_track_local() {
     assert(!snat_v4_track_local(&ctx, &tuple, &state, NAT_DIR_EGRESS, 0,
                                 &target));
 }
+
+// Create a fake map
+HASHMAP(struct ipv4_ct_tuple, struct ipv4_nat_entry) ipv4_ct_tuple_map;
+
+// Define the compare_func
+// We need to wrap the cmp func upon using since memcmp requires the length.
+int bpf_compare(const void *a, const void *b)
```

Comment:
Is there any reason these aren't static? e.g. I'd guess we want multiple tests compiled into one binary and we may not want these polluting the global namespace.

Reply:
Thanks for pointing it out. Since func `memcmp` requires a argument `size` which indicates the number of bytes to be compared, we cannot define the compare function until we know the size of the key in a particular fake eBPF map (in this example the key is of type `struct ipv4_ct_tuple`). If there is another test using user-space map emulation, we need to define a new set of these function for it. So I think it does not matter if they are static or not?

Reply:
I just realized that `bpf_compare` and `bpf_hash` might be confusing, so I changed them to `bpf_compare_ipv4_ct_tuple` and `bpf_hash_ipv4_ct_tuple` which indicate that these functions are for key `struct ipv4_ct_tuple` only.